python/ray/monitor.py

```diff
@@ -478,6 +482,104 @@ def driver_removed_handler(self, unused_channel, data):
 
         self._clean_up_entries_for_driver(driver_id)
 
+    def _xray_clean_up_entries_for_driver(self, driver_id):
+        """Remove this driver's object/task entries from redis.
+
+        Removes control-state entries of all tasks and task return
+        objects belonging to the driver.
+
+        Args:
+            driver_id: The driver id.
+        """
+        redis = self.state.redis_clients[0]
+
+        xray_task_table_prefix = (
+            ray.gcs_utils.TablePrefix_RAYLET_TASK_string.encode("ascii"))
+        xray_object_table_prefix = (
+            ray.gcs_utils.TablePrefix_OBJECT_string.encode("ascii"))
+
+        task_table_infos = {}  # task id -> TaskInfo
+        for key in redis.scan_iter(match=xray_task_table_prefix + b"*"):
+            entry = redis.get(key)
```

Comment:
Instead of directly using `redis.get` we should probably use the `RAY.TABLE_LOOKUP` interface for doing this. We should also use `state._execute_command` to execute Redis commands so that they go to the correct shard.

Reply:
Similarly `state._keys` instead of `redis.scan_iter`.

Though we should probably replace the `keys` in https://github.com/ray-project/ray/blob/8e687cbc9838fce0b1d8190c7851b7d573562487/python/ray/experimental/state.py#L171 with `scan_iter` like you're using here.

Reply:
This might make sense to do in a subsequent PR, when we enable the shard-related tests in `monitor_test.py`.

Reply:
Hm, well it's a simple change (just a few substitutions), and then it will automatically continue to work when sharding is implemented.